scenes/game/local_game.gd

```diff
@@ -4318,6 +4333,94 @@ func handle_strike_effect(card_id :int, effect, performing_player : Player):
 				# match card.definition['id']'s instead.
 				named_card_name = named_card.definition['id']
 			events += opposing_player.discard_matching_or_reveal(named_card_name)
+		"name_range":
+			decision_info.clear()
+			decision_info.type = Enums.DecisionType.DecisionType_PickNumberFromRange
+			decision_info.player = performing_player.my_id
+			decision_info.choice_card_id = card_id
+			decision_info.choice = []
+			decision_info.limitation = []
+			if effect['target_effect'] == "opponent_discard_range_or_reveal":
+				decision_info.amount_min = 0
+				decision_info.amount = 9
+				decision_info.valid_zones = ["Range X", "Range -"]
+				decision_info.effect_type = "have opponent discard a card including that Range or reveal their hand"
+				for i in range(decision_info.amount + 1):
+					decision_info.limitation.append(i)
+					decision_info.choice.append({
+						"effect_type": "opponent_discard_range_or_reveal",
+						"target_range": i,
+						"amount": 1
+					})
+				var next_num = decision_info.amount + 1
+				for i in range(2):
+					decision_info.limitation.append(next_num)
+					decision_info.choice.append({
+						"effect_type": "opponent_discard_range_or_reveal",
+						"target_range": decision_info.valid_zones[i],
+						"amount": 1
+					})
+
+				change_game_state(Enums.GameState.GameState_PlayerDecision)
+				events += [create_event(Enums.EventType.EventType_PickNumberFromRange, performing_player.my_id, 0)]
+			else:
+				assert(false, "Target effect for name_range not found.")
+				decision_info.clear()
+		"opponent_discard_range_or_reveal":
+			var target_range = effect['target_range']
+			var card_ids_in_range = []
+			if target_range is String:
+				if target_range == "Range X":
+					# If the range is a string like "TOTAL_POWER".
+					for card in opposing_player.hand:
+						if card.definition['range_min'] is String or card.definition['range_max'] is String:
+							card_ids_in_range.append(card.id)
+				elif target_range == "Range -":
+					# If the range is -1 like Block.
+					for card in opposing_player.hand:
+						var card_range = card.definition['range_min']
+						if is_number(card_range) and card_range == -1:
+							card_ids_in_range.append(card.id)
+				else:
+					assert(false, "Unknown target range")
+			else:
+				# If the range is an actual number.
+				for card in opposing_player.hand:
+					var card_range_min = card.definition['range_min']
+					var card_range_max = card.definition['range_max']
+					# Some cards have mixed numbers and strings.
+					# Potentially evaluate? Currently only Phonon though and max==min if not in strike.
```

Comment:
If you name, for example, 3 when you're at range 3 against Nine, would Kunzite be considered in range? or does that only get targeted if you name Range X? if the former, it might be helpful to attempt to resolve the ranges (maybe the range resolution function should have logic similar to this if a strike isn't active, in either case)